crates/sui-core/src/signature_mpc/identifiable_abort.rs

```diff
@@ -0,0 +1,181 @@
+};
+
+use crate::authority::authority_per_epoch_store::AuthorityPerEpochStore;
+use crate::signature_mpc::sign::{SignRound, SignRoundCompletion};
+use crate::signature_mpc::sign::SignState;
+use crate::signature_mpc::submit_to_consensus::SubmitSignatureMPC;
+
+/// Generate a list of proofs, one for every message in the messages batch that its decryption failed.
+/// Each proof proves that the executing party id (state.party_id) while signing on that message.
+pub fn generate_proofs(
+    state: &SignState,
+    failed_messages_indices: &Vec<usize>,
+) -> Vec<(
+    PartialDecryptionProof,
+    SignaturePartialDecryptionProofVerificationParty,
+)> {
+    let decryption_key_share = DecryptionKeyShare::new(
+        state.party_id,
+        state.tiresias_key_share_decryption_key_share,
+        &state.tiresias_public_parameters,
+    )
+    .unwrap();
+
+    failed_messages_indices
+        .iter()
+        .map(|index| {
+            generate_proof(
+                state.tiresias_public_parameters.clone(),
+                decryption_key_share.clone(),
+                state.party_id,
+                state.presigns.clone().unwrap().get(*index).unwrap().clone(),
+                state
+                    .tiresias_public_parameters
+                    .encryption_scheme_public_parameters
+                    .clone(),
+                state
+                    .public_nonce_encrypted_partial_signature_and_proofs
+                    .clone()
+                    .unwrap()
+                    .get(*index)
+                    .unwrap()
+                    .clone(),
+            )
+        })
+        .collect()
+}
+
+/// Identify all the parties that behaved maliciously in this messages batch.
+pub(crate) fn identify_batch_malicious_parties(
+    state: &SignState,
+) -> twopc_mpc_protocols::Result<HashSet<PartyID>> {
+    // Need to call [`generate_proofs`] to re-generate the SignaturePartialDecryptionProofVerificationParty objects,
+    // that are necessary to call the [`identify_malicious_parties`] function.
+    let failed_messages_parties = generate_proofs(&state, &state.failed_messages_indices.clone().unwrap());
+    let mut malicious_parties = HashSet::new();
+    let involved_shares = get_involved_shares(&state);
+    for ((i, message_index), (_, party)) in state
+        .clone()
+        .failed_messages_indices
+        .unwrap()
+        .into_iter()
+        .enumerate()
+        .zip(failed_messages_parties.into_iter())
+    {
+        let (shares, masked_shares) = change_shares_type(&involved_shares, message_index);
+        let involved_proofs = get_involved_proofs(&state, i);
+        identify_message_malicious_parties(
+            party,
+            shares,
+            masked_shares,
+            state.tiresias_public_parameters.clone(),
+            involved_proofs,
+            state
+                .involved_parties
+                .as_deref()
+                .unwrap_or(&Vec::new())
```

Comment:
```suggestion
                .unwrap_or(&Vec::new())
```
What is this? Please be mindful about error handling. This is definitely not the way the error should be handeled.